### DISCUSSION TITLE
[FEATURE REQUEST] add filter from api in datatable

### DISCUSSION THREAD

**Discussion by zulfikar-ditya** [filamentphp/filament#7639] [Ideas]
in filament table maybe should have a filter feature, but the data not from array or enums. it from api

let say we have data category (10,000) categories, we will show all the categories in the filter, not using limit or paginate. show we need to build an api that retrieve the list of the categories .

it's that any option about this features?

**Comment by pboivin** [comment]
This should be possible with the SelectFilter or a custom filter form:
https://filamentphp.com/docs/3.x/tables/filters#select-filters

**Comment by zulfikar-ditya** [comment]
@pboivin yeah i already read about that feature you mention. maybe filament team should add this features in next updates

**Reply by danharrin** [reply]
There is no way we can make it easier than it already is, I don't know what you're expecting us to do

**Reply by zulfikar-ditya** [reply]
i know about that. it maybe take long time or complex changes because the filament teams already build as general as possible.

**Comment by margarizaldi** [comment]
> i know about that. it maybe take long time or complex changes because the filament teams already build as general as possible.

Bukan soal rumit/lamanya bang, tapi yg ada udah sangat cukup dan apa yg abang sampaikan udah tercover, apa yg belum bisa coba? mau ngeload 2jt kategori pun ga perlu api-apian
